storage/interface.go

```diff
@@ -285,6 +285,12 @@ type Appender interface {
 	CreatedTimestampAppender
 }
 
+type AppenderWithCT interface {
+	Appender
+
+	AppendWithCT(ref SeriesRef, l labels.Labels, t int64, v float64, ct int64) (SeriesRef, error)
```

Comment:
I added an alternative implementation that doesn't add an interface in https://github.com/prometheus/prometheus/pull/15255 but its uglier IMO. I had another draft that tried to use options to do this but import cycles and other issues made that approach untennable.

cc @bwplotka @ArthurSens